[Queues] Overview page (#8679)
cloudflare/cloudflare-docs: content/queues/_index.md

type: overview
pcx_content_type: overview
weight: 1
layout: overview
meta:
  title: Cloudflare Queues
---

# Cloudflare Queues

{{<description>}}

> Cloudflare Queues allows developers to send and receive messages with guaranteed delivery. 

Reviewer: ```suggestion
Send and receive messages with guaranteed delivery and no charges for egress bandwidth.
```

Reviewer: Adjusted for copywriting best practices.

{{</description>}}

{{<plan type="paid">}}

{{<Aside>}}

[Cloudflare Queues is now in open Beta](https://blog.cloudflare.com/cloudflare-queues-open-beta/). Join the [`#queues-beta`](https://discord.gg/rrZXVVcKQF) channel in our Developer Discord to learn more.

{{</Aside>}}

> Cloudflare Queues enable you to build applications that can [guarantee delivery](/queues/learning/delivery-guarantees/), offload work from a request, [send data from Worker to Worker](/queues/learning/consumer-concurrency/), [buffer or batch data](/queues/learning/batching-retries/).

Reviewer: ```suggestion
Cloudflare Queues integrates with [Cloudflare Workers](/workers/) and enables you to build applications that can [guarantee delivery](/queues/learning/delivery-guarantees/), offload work from a request, [send data from Worker to Worker](/queues/platform/configuration/), and [buffer or batch data](/queues/learning/batching-retries/).
```

Reviewer: Concurrency is not related to sending data from Worker to Worker, so updating the link to relevant place.

Reviewer: You can ask Charlie what part of the documentation is relevant to offloading work from a request but I think it might be this: developers.cloudflare.com/queues/platform/javascript-apis/#consumer

- **Buffer or batch data**. If a service has bursty load, you can send that load to a Queue, then process it in predictable batches.

{{</content-column>}}

---

## Features

{{<feature header="Wrangler" href="/workers/wrangler/install-and-update/">}}
The Workers command-line interface, Wrangler, allows you to [create](/workers/wrangler/commands/#init), [test](/workers/wrangler/commands/#dev), and [deploy](/workers/wrangler/commands/#publish) your Workers projects.
{{</feature>}}

---

## Related products
 
{{<related header="R2" href="/r2/" product="r2">}}

Cloudflare R2 Storage allows developers to store large amounts of unstructured data without the costly egress bandwidth fees associated with typical cloud storage services.

{{</related>}}

{{<related header="Workers" href="/workers/" product="workers">}}

Cloudflare Workers allows developers to build serverless applications and deploy instantly across the globe for exceptional performance, reliability, and scale.

{{</related>}}

---

## More resources

{{<resource-group>}}

{{<resource header="Plans" href="/workers/platform/pricing/" icon="price">}}Learn about Free and Paid plans.{{</resource>}}

> {{<resource header="Limits" href="/workers/platform/limits/" icon="documentation-clipboard">}}Learn about plan limits (Free plans get 100,000 requests per day).{{</resource>}}

Reviewer: These links point to Workers docs and they need to point to Queues docs.
